src/integration/java/io/pinecone/helpers/IndexManager.java

```diff
@@ -12,69 +11,79 @@
-        if(indexName.isEmpty()) indexName = createNewIndex(dimension, controlPlaneClient);
+        String indexName = findIndexWithDimensionAndType(indexList, dimension, controlPlaneClient, indexType);
+        if (indexName.isEmpty()) indexName = createNewIndex(controlPlaneClient, indexType, dimension);
 
         // Do not proceed until the newly created index is ready
         isIndexReady(indexName, controlPlaneClient);
         PineconeClient dataPlaneClient = new PineconeClient(config);
-        IndexMeta indexMeta = controlPlaneClient.describeIndex(indexName);
-        String host = indexMeta.getStatus().getHost();
+        String host = controlPlaneClient.describeIndex(indexName).getHost();
 
         return dataPlaneClient.connect(
                 new PineconeConnectionConfig()
                         .withConnectionUrl("https://" + host));
     }
 
-    public static String createIndexIfNotExistsControlPlane(PineconeClientConfig config, int dimension) throws IOException, InterruptedException {
-        PineconeIndexOperationClient controlPlaneClient = new PineconeIndexOperationClient(config);
-        List<String> indexList = controlPlaneClient.listIndexes();
-        String indexName = findIndexWithDimensionAndPodType(indexList, dimension, controlPlaneClient);
+    public static String createIndexIfNotExistsControlPlane(PineconeIndexOperationClient controlPlaneClient, int dimension, String indexType) throws IOException, InterruptedException {
+        IndexList indexList = controlPlaneClient.listIndexes();
+        String indexName = findIndexWithDimensionAndType(indexList, dimension, controlPlaneClient, indexType);
 
-        return (indexName.isEmpty()) ? createNewIndex(dimension, controlPlaneClient) : indexName;
+        return (indexName.isEmpty()) ? createNewIndex(controlPlaneClient, indexType, dimension) : indexName;
     }
 
-    private static String findIndexWithDimensionAndPodType(List<String> indexList, int dimension, PineconeIndexOperationClient controlPlaneClient)
-            throws IOException, InterruptedException {
+    private static String findIndexWithDimensionAndType(IndexList indexList, int dimension, PineconeIndexOperationClient controlPlaneClient, String indexType)
+            throws InterruptedException {
         int i = 0;
-        while (i < indexList.size()) {
-            IndexMeta indexMeta = isIndexReady(indexList.get(i), controlPlaneClient);
-            if (indexMeta.getDatabase().getDimension() == dimension
-                    && (indexMeta.getDatabase().getPodType().equals("p1.x1") || config.getEnvironment().equals("gcp-starter"))) {
-                return indexList.get(i);
+        List<IndexModel> indexModels = indexList.getIndexes();
+        while (i < indexModels.size()) {
+            IndexModel indexModel = isIndexReady(indexModels.get(i).getName(), controlPlaneClient);
+            // ToDo: add pod type support
```

Comment:
In the future PR, I'd like to test for gcp-starter s.t. no new indexes are created if the gcp-starter is detected.